199.py

```diff
@@ -0,0 +1,36 @@
+from typing import List
+from collections import deque
+
+# Definition for a binary tree node.
+# class TreeNode:
+#     def __init__(self, val=0, left=None, right=None):
+#         self.val = val
+#         self.left = left
+#         self.right = right
+class Solution:
+    def rightSideView(self, root: TreeNode) -> List[int]:
+        right_nodes = []
+        
+        if not root:
+            return right_nodes
+        
+        queue = deque([root])
+        
+        while queue:
+            level_length = len(queue)
+            
+            while level_length > 0:
+                node = queue.popleft()
+                
+                if level_length == 1:
+                    right_nodes.append(node.val)
+                    
+                if node.left:
+                    queue.append(node.left)
+                    
+                if node.right:
+                    queue.append(node.right)
+                    
+                level_length -= 1
```

Comment:
Same as [this](https://github.com/woke7/leetcode-python/pull/19#discussion_r606710471)

Reply:
This one is a bit different, as I need `level_length` in each iteration to determine if that the rightmost node at that level.

Reply:
You can still use a `for` loop:

```python
for i in range(level_length):
    if i == level_length - 1:
        # append your right-most node's value here
```

What's good about this `for` loop is that it avoids the manual maintenance of `level_length`.